postgis/ewkb.py

```diff
@@ -13,15 +13,20 @@ def __new__(mcs, name, bases, attrs, **kwargs):
             Typed.types[cls.TYPE] = cls
         return cls
 
+    def __call__(cls, *args, **kwargs):
+        # Allow to pass an instance as first argument, for blind casting.
+        if args and isinstance(args[0], cls):
+            return args[0]
+        return super().__call__(*args, **kwargs)
 
-class EWKBReader(object):
-    # Should we use shapely/libgeos instead?
+
+class Reader:
 
     def __init__(self, stream):
         self.stream = stream
 
     def clone(self):
-        return EWKBReader(self.stream)
+        return type(self)(self.stream)
```

Comment:
Dynamically return current class, so if this get inherited it will return the inherited class (I may have done it that way too: `self.__class__()`).